src/commands/registries/azure/deleteAzureRegistry.ts

```diff
@@ -3,27 +3,33 @@
  *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DialogResponses, IActionContext } from '@microsoft/vscode-azext-utils';
-import { l10n, ProgressLocation, window } from 'vscode';
+import { DialogResponses, IActionContext, contextValueExperience } from '@microsoft/vscode-azext-utils';
+import { ProgressLocation, l10n, window } from 'vscode';
 import { ext } from '../../../extensionVariables';
-import type { AzureRegistryTreeItem } from '../../../tree/registries/azure/AzureRegistryTreeItem';
-import { registryExpectedContextValues } from '../../../tree/registries/registryContextValues';
+import { AzureRegistry, AzureRegistryDataProvider } from '../../../tree/registries/Azure/AzureRegistryDataProvider';
+import { UnifiedRegistryItem } from '../../../tree/registries/UnifiedRegistryTreeDataProvider';
 
-export async function deleteAzureRegistry(context: IActionContext, node?: AzureRegistryTreeItem): Promise<void> {
+export async function deleteAzureRegistry(context: IActionContext, node?: UnifiedRegistryItem<AzureRegistry>): Promise<void> {
     if (!node) {
-        node = await ext.registriesTree.showTreeItemPicker<AzureRegistryTreeItem>(registryExpectedContextValues.azure.registry, { ...context, suppressCreatePick: true });
+        // we can't pass in the azure tree provider because it's not a UnifiedRegistryItem and we need the provider to delete
+        node = await contextValueExperience(context, ext.registriesTree, { include: 'azureContainerRegistry' });
```

Comment:
You can though right? Pass in the Azure registry provider here, deal with the wrapper/not-a-wrapper thing, and when it's time to delete just use the Azure provider again